endpoint/endpoint.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/select.h>
+#include "http_parser.h"
+#include "utils/cipher.h"
+#include "utils/connectivity/conn_http.h"
+#include "utils/https.h"
+#include "utils/text_serializer.h"
+#include "utils/tryte_byte_conv.h"
+#include "utils/trytes.h"
+
+#define HOST "tangle-accel.puyuma.org"
+#define PORT "443"
+#define API "transaction/"
+#define SSL_SEED "nonce"
+#define ADDRESS                                                                \
+  "POWEREDBYTANGLEACCELERATOR999999999999999999999999999999999999999999999999" \
+  "999999A"
+#define ADDR_LEN 81
+#define MAX_MSG_LEN 1024
+
+static char addr_log_template[] = "\n%s\n";
+
+int log_address(uint8_t *next_addr) {
+  FILE *fp;
+  char addr_log[ADDR_LEN + 3];
+  // Append the next address to the address log file
+  fp = fopen(ADDR_LOG_PATH, "a");
+  if (!fp) {
+    perror("open addr_log.log failed:");
+    fclose(fp);
```

Comment:
If it fails to open given file(s), we shall not close the fd since it is `NULL`.